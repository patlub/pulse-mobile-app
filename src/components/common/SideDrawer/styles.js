import { StyleSheet, Platform } from 'react-native'

const styles = StyleSheet.create({
  drawer: {
    flex: 1,
    flexDirection: 'column',
  },
  header: {
    flex: 0.30,
    paddingTop: Platform.OS === 'ios' ? 30 : 10,
    backgroundColor: '#2443DA',
    flexDirection: 'column',
  },
  headerContent: {
    flex: 1,
    justifyContent: 'space-between',
  },
  profileStyle: {
    flex: 0.9,
    flexDirection: 'column',
    marginStart: 15,
    paddingTop: 10,
  },
  avatarStyle: {
    height: 90,
    width: 90,
    borderRadius: 45,
    resizeMode: 'cover',
  },
  titleStyle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'white',
    paddingTop: 30,
  },
  mailContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-evenly',
    paddingBottom: 5,
  },
  mailText: {
    color: 'white',
    fontSize: 15,
  },
  iconStyle: {
    color: 'white',
    marginStart: 15,
  },
  iconRotate: {
    transform: [{ rotate: '180deg' }],
  },
  navContainer: {
    flex: 0.70,
    marginTop: 10,
    marginStart: 20,
  },
  partnersNav: {
    flexDirection: 'row',
    paddingVertical: 10,
  },
  partnersLabel: {
    color: '#4285f4',
    fontSize: 15,
    fontWeight: 'bold',
    marginStart: 20,
    alignSelf: 'center',
  },
  partnersIcon: {
    backgroundColor: '#2443DA',
    height: 45,
    width: 45,
    borderWidth: 1,
    borderColor: '#2443DA',
    borderRadius: 5,
  },
  newCheckinNav: {
    flexDirection: 'row',
    paddingVertical: 10,
  },
  newCheckinIcon: {
    backgroundColor: '#2443DA',
    height: 45,
    width: 45,
    borderWidth: 1,
    borderColor: '#2443DA',
    borderRadius: 5,
  },
  newCheckinLabel: {
    color: '#4285f4',
    fontSize: 15,
    fontWeight: 'bold',
    marginStart: 20,
    alignSelf: 'center',
  },
  logoutNav: {
    flexDirection: 'row',
    paddingVertical: 10,
  },
  logoutIcon: {
    backgroundColor: '#2443DA',
    height: 45,
    width: 45,
    borderWidth: 1,
    borderColor: '#2443DA',
    borderRadius: 5,
  },
  logoutLabel: {
    color: '#4285f4',
    fontSize: 15,
    fontWeight: 'bold',
    marginStart: 20,
    alignSelf: 'center',
  },
})

export default styles
