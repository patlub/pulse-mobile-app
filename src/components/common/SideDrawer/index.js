import React from 'react'
import { Text, View, Image, TouchableOpacity } from 'react-native'
import { Icon, Avatar } from 'react-native-elements'


import styles from './styles'

class SideDrawer extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      showLogout: true,
    }
  }

  toggleShowLogout(showLogout) {
    this.setState({ showLogout: !showLogout })
  }

  render() {
    return (
      <View style={styles.drawer}>
        <View style={styles.header}>
          <View style={styles.headerContent}>
            <View style={styles.profileStyle}>
              <Image
                style={styles.avatarStyle}
                source={{
                  uri: 'https://raw.githubusercontent.com/sirjmkitavi/gui/master/static/profile.jpg',
                  cache: 'only-if-cached',
              }}
              />
              <Text style={styles.titleStyle}>
                  Test Admin
              </Text>
            </View>
            <TouchableOpacity
              style={styles.mailContainer}
              onPress={() => this.toggleShowLogout(this.state.showLogout)}
            >
              <Text style={styles.mailText}>
                  test-user-admin@andela.com
              </Text>
              <Icon
                name="arrow-drop-down"
                iconStyle={[styles.iconStyle, this.state.showLogout ? styles.iconRotate : null]}
              />
            </TouchableOpacity>
          </View>
        </View>

        <View style={styles.navContainer}>

          {!this.state.showLogout ? (
            <View>
              <TouchableOpacity style={styles.partnersNav}>
                <Avatar
                  medium
                  icon={{
                    name: 'people',
                    type: 'simple-line-icon',
                  }}
                  overlayContainerStyle={styles.partnersAvatar}
                />
                <Text style={styles.partnersLabel}>
                  Partners
                </Text>
              </TouchableOpacity>
              <TouchableOpacity style={styles.newCheckinNav}>
                <Avatar
                  medium
                  icon={{
                    name: 'plus',
                    type: 'feather',
                  }}
                  overlayContainerStyle={styles.newCheckinAvatar}
                />
                <Text style={styles.newCheckinLabel}>
                  New Checkin
                </Text>
              </TouchableOpacity>
            </View>
          ) : (
            <View>
              <TouchableOpacity style={styles.partnersNav}>
                <Avatar
                  medium
                  icon={{
                    name: 'logout-variant',
                    type: 'material-community',
                  }}
                  overlayContainerStyle={styles.partnersAvatar}
                />
                <Text style={styles.partnersLabel}>
                  Log Out
                </Text>
              </TouchableOpacity>
            </View>
          )}

        </View>

      </View>
    )
  }
}

export default SideDrawer
