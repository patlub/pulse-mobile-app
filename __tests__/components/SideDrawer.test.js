import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import SideDrawer from '../../src/components/common/SideDrawer'

describe('Testing SideDrawer Component', () => {

  navigation = {
    navigate: jest.fn(),
  }

  it('renders as expected', () => {
    const wrapper = shallow(<SideDrawer navigation={navigation}/>)
    expect(wrapper).toMatchSnapshot();
  })

  it('toggleShowLogout: should set state to new state', () => {
    const wrapper = shallow(<SideDrawer navigation={navigation}/>)
    const emailButton = wrapper.find('TouchableOpacity').at(0)
    emailButton.simulate('Press')
    expect(wrapper.state().showLogout).toBe(true)
  })

  it('navigates to Partners screen', () => {
    const wrapper = shallow(<SideDrawer navigation={navigation}/>)
    const partnersNav = wrapper.find('TouchableOpacity').at(1)
    partnersNav.simulate('Press');
    expect(navigation.navigate).toBeCalled()
  })

  it('navigates to checkin screen', () => {
    const wrapper = shallow(<SideDrawer navigation={navigation}/>)
    const checkinNav = wrapper.find('TouchableOpacity').at(2)
    checkinNav.simulate('Press');
    expect(navigation.navigate).toBeCalled()
  })

  it('navigates to logout', () => {
    const wrapper = shallow(<SideDrawer navigation={navigation}/>)
    wrapper.setState({showLogout: true})
    const logoutNav = wrapper.find('TouchableOpacity').at(1)
    logoutNav.simulate('Press');
    expect(navigation.navigate).toBeCalled()
  })
})
