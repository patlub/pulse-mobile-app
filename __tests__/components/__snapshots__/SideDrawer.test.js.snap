// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SideDrawer Component renders as expected 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#2443DA",
        "flex": 0.3,
        "flexDirection": "column",
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0.9,
            "flexDirection": "column",
            "marginStart": 15,
            "paddingTop": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://raw.githubusercontent.com/sirjmkitavi/gui/master/static/profile.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 45,
              "height": 90,
              "resizeMode": "cover",
              "width": 90,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
              "paddingTop": 30,
            }
          }
        >
          Test Admin
        </Text>
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "paddingBottom": 5,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 15,
            }
          }
        >
          test-user-admin@andela.com
        </Text>
        <Icon
          color="black"
          iconStyle={
            Array [
              Object {
                "color": "white",
                "marginStart": 15,
              },
              null,
            ]
          }
          name="arrow-drop-down"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.7,
        "marginStart": 20,
        "marginTop": 10,
      }
    }
  >
    <View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "paddingVertical": 10,
          }
        }
      >
        <Avatar
          editButton={
            Object {
              "iconColor": "#fff",
              "iconName": "mode-edit",
              "iconType": "material",
              "size": null,
              "style": null,
              "underlayColor": "#000",
            }
          }
          icon={
            Object {
              "name": "people",
              "type": "simple-line-icon",
            }
          }
          medium={true}
          onEditPress={null}
          overlayContainerStyle={
            Object {
              "backgroundColor": "#2443DA",
              "borderColor": "#2443DA",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 45,
              "width": 45,
            }
          }
          showEditButton={false}
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#4285f4",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginStart": 20,
            }
          }
        >
          Partners
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "paddingVertical": 10,
          }
        }
      >
        <Avatar
          editButton={
            Object {
              "iconColor": "#fff",
              "iconName": "mode-edit",
              "iconType": "material",
              "size": null,
              "style": null,
              "underlayColor": "#000",
            }
          }
          icon={
            Object {
              "name": "plus",
              "type": "feather",
            }
          }
          medium={true}
          onEditPress={null}
          overlayContainerStyle={
            Object {
              "backgroundColor": "#2443DA",
              "borderColor": "#2443DA",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 45,
              "width": 45,
            }
          }
          showEditButton={false}
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#4285f4",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginStart": 20,
            }
          }
        >
          New Checkin
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
